src/mpid/ch4/src/ch4_types.h

```diff
@@ -82,6 +82,17 @@ typedef struct MPIDIG_send_long_lmt_msg_t {
     MPIR_Request *rreq_ptr;
 } MPIDIG_send_long_lmt_msg_t;
 
+typedef struct MPIDIG_send_gpu_ipc_req_msg_t {
+    MPIDIG_hdr_t hdr;
+    size_t data_sz;
+    MPL_gpu_ipc_mem_handle_t mem_handle;
+    MPIR_Request *sreq_ptr;
+} MPIDIG_send_gpu_ipc_req_msg_t;
+
+typedef struct MPIDIG_send_gpu_ipc_ack_msg_t {
+    MPIR_Request *sreq_ptr;
+} MPIDIG_send_gpu_ipc_ack_msg_t;
+
```

Comment:
I know the legacy code does this, but these two structures are between the send function in `mpidig_am_gpu.h` and the corresponding target callback functions in `mpidig_am_gpu.c`, it would be nice to keep the definition inside rather than expose globally. Can we move them to `mpidig_am_gpu.h`, for example?